syntax="proto3";//声明protobuf版本

package fixbug;//声明了代码所在的包（对于C++来说是namespace）

//定义下面选项，表示生成service服务类和rpc方法描述，因为是默认不生成
option cc_generic_services=true;

message Resultcode{
    int32 errcode=1;
    bytes errmsg=2;
}

//定义登录请求消息类型 name pwd
message LoginRequest{
    bytes name=1;//定义protobuf第一个字段
    bytes pwd=2;//定义protobuf第二个字段
}

//定义登录响应消息类型
message LoginResponse{
    Resultcode result=1;
    bool success=2;

}

message GetFriendListsRequest{
    uint32 userid=1;
}

message User{
    bytes name=1;
    uint32 age=2;
    enum Sex{ //枚举
        MAN =0;
        WOMAN=1;
    }
    Sex sex=3;
}

message GetFriendListsResponse{
    Resultcode result=1;
    repeated User frined_list=2;//定义一个列表类型
}

//在protobuf中定义描述rpc方法的类型 -service
service UserServiceRpc{
    rpc Login(LoginRequest) returns(LoginResponse); 
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}